### PR TITLE
Receive error body from Run Service

### PR DESCRIPTION
**Pull request by ericsciple**
Receive error body from Run Service so we can distinguish responses from proxies/etc

On an E2E Codespace, I tested an error response from each affected REST API (acquire job, complete job, renew job)

### CODE REVIEW

File: src/Runner.Common/RunServer.cs [hunk 1]
```diff
@@ -62,7 +62,9 @@ public Task<AgentJobRequestMessage> GetJobMessageAsync(string id, CancellationTo
             CheckConnection();
             return RetryRequest<AgentJobRequestMessage>(
                 async () => await _runServiceHttpClient.GetJobMessageAsync(requestUri, id, VarUtil.OS, cancellationToken), cancellationToken,
-                shouldRetry: ex => ex is not TaskOrchestrationJobAlreadyAcquiredException);
+                shouldRetry: ex => ex is not TaskOrchestrationJobNotFoundException &&
```
Comment (ericsciple) [inline src/Runner.Common/RunServer.cs:65]: revert for now

File: src/Sdk/RSWebApi/RunServiceHttpClient.cs [hunk 2]
```diff
@@ -86,21 +86,43 @@ static RunServiceHttpClient()
...
                 content: requestContent,
+                readErrorContent: true,
                 cancellationToken: cancellationToken);
 
             if (result.IsSuccess)
             {
                 return result.Value;
             }
 
+            if (TryParseErrorContent(result.ErrorContent, out RunServiceError error))
```
Comment (ericsciple) [inline src/Sdk/RSWebApi/RunServiceHttpClient.cs:97]: apply to the rest of run service

File: src/Sdk/RSWebApi/RunServiceHttpClient.cs [hunk 3]
```diff
@@ -169,19 +208,37 @@ static RunServiceHttpClient()
...
             {
                 case HttpStatusCode.NotFound:
                     throw new TaskOrchestrationJobNotFoundException($"Job not found: {jobId}");
-                default:
-                    throw new Exception($"Failed to renew job: {result.Error}");
+            }
+
+            if (!string.IsNullOrEmpty(result.ErrorBody))
+            {
+                throw new Exception($"Failed to renew job: {result.Error}. {result.ErrorBody}");
```
Comment (ericsciple) [inline src/Sdk/RSWebApi/RunServiceHttpClient.cs:237]: todo: truncate ErrorBody if greater than 1000 chars

File: src/Sdk/WebApi/WebApi/RawHttpClientBase.cs [hunk 4]
```diff
@@ -101,15 +101,55 @@ public void Dispose()
             }
         }
 
+        protected async Task<RawHttpClientResult> Send2Async(
```
Comment (yaananth) [inline src/Sdk/WebApi/WebApi/RawHttpClientBase.cs:104]: Should we better name `Send2Async`? This is RawHttp, so not every call needs to have json response like run-service does, in that sense, we might want to keep `SendAsync` and this new one as well?

**Review by yaananth — APPROVED** [review]

File: src/Sdk/WebApi/WebApi/RawHttpClientBase.cs (hunk 4 above)
Comment (yaananth) [reply src/Sdk/WebApi/WebApi/RawHttpClientBase.cs:104]: or since we have `try*` anyway, why do we need `send2`? we plan to delete `send` one later? 🤔

**Review by TingluoHuang — APPROVED** [review]

File: src/Sdk/WebApi/WebApi/RawHttpClientBase.cs (hunk 4 above)
Comment (ericsciple) [reply src/Sdk/WebApi/WebApi/RawHttpClientBase.cs:104]: Open to changing this. I played around with a few different names but everything was horrible.

E.g. "SendAndReceiveWrapperResult". Such a long, aesthetically displeasing name.

**Pull request merged**